core/templates/domain/suggestion/suggestion.model.ts

```diff
@@ -74,30 +70,24 @@ export class Suggestion {
     this.lastUpdatedMsecs = lastUpdatedMsecs;
   }
 
-  getThreadId(): string {
-    return this.threadId;
-  }
-}
-
-@Injectable({providedIn: 'root'})
-export class SuggestionObjectFactory {
-  constructor(private suggestionsService: SuggestionsService) {}
-  createFromBackendDict(
+  static createFromBackendDict(
       suggestionBackendDict: SuggestionBackendDict): Suggestion {
-    let threadId = this.suggestionsService.getThreadIdFromSuggestionBackendDict(
+    let threadId = SuggestionsService.getThreadIdFromSuggestionBackendDict(
```

Comment:
This won't work. Since `getThreadIdFromSuggestionBackendDict` is not a static member of the SuggestionService. Can you instead change this to
```suggestion
    let threadId = suggestionBackendDict.suggestion_id;
```
This is the reason why your tests are failing at the moment.

Reply:
Thanks for the explanation . Done!